### PR TITLE
Fix read of vcd_vapp_vm resource when VM it isn't found. Allows terraform to recreate resource and not fail

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Ref: https://github.com/vmware/terraform-provider-vcd/issues/611

### CODE REVIEW

**Review by adambarreiro — APPROVED** [review]

File: .changes/v3.6.0/783-bug-fixes.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Fix Issue #611 when read of `vcd_vapp_vm` resource failed when vm isn't found. Now allows terraform to recreate resource when it isn't found. [GH-783]
```
[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply .changes/v3.6.0/783-bug-fixes.md:1]: thank you, updated

**Review by lvirbalas — COMMENTED** [review]
What about adding a test for this use case?

**Comment by vbauzysvmware** [comment]
> What about adding a test for this use case?

**Comment by vbauzysvmware** [comment]
> What about adding a test for this use case?

Not sure it's possible to check the result, will check.

**Review by Didainius — APPROVED** [review]
Fantastic :) Tried on both `vcd_vapp_vm` and `vcd_vm` - works great.

File: .changes/v3.6.0/783-bug-fixes.md [hunk 2]
```diff
@@ -0,0 +1 @@
+* Fix Issue #611 when read of `vcd_vapp_vm` resource failed when VM isn't found. Now allows Terraform to recreate resource when it isn't found. [GH-783]
```
[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply .changes/v3.6.0/783-bug-fixes.md:1]: added

**Review by mikeletux — APPROVED** [review]
Run both `vcd_vapp_vm` and `vcd_vm`. 
When applying it for the second time when previously deleted the vm, it recreates it.

Thanks for the effor @vbauzysvmware

**Pull request merged**